module/Kbuild.in

```diff
@@ -472,3 +472,12 @@ OBJECT_FILES_NON_STANDARD_vdev_raidz_math_avx512f.o := y
 ifeq ($(CONFIG_ALTIVEC),y)
 $(obj)/zfs/vdev_raidz_math_powerpc_altivec.o : c_flags += -maltivec
 endif
+
+# Woe betide ye who override this, for yours is the kingdom of mysterious segfaults
+# absolutely everywhere.
+#
+# Free warning to anyone considering it - -march=native or the like later in the line
+# will undo -mgeneral-regs-only, and gcc's -O2 starting in 12 does autovectorizing.
+#
+# Good luck.
+ZFS_MODULE_CFLAGS += -mgeneral-regs-only -fno-tree-vectorize
```

Comment:
Perhaps I'm just overlooking it, but don't we also need to enforce this for the user space build?

Reply:
Thanks, I missed that hunk in my git add, apparently.

Will push shortly - had my primary machine die yesterday, still getting set up on the replacement one...

Reply:
Actually, wait, depends what behavior we want to enforce.

`-fno-tree-vectorize` makes sense to enforce in userland too, but `-mgeneral-regs-only` seems unnecessary since we nop out the fpu save/restore dance as irrelevant.

I'm fine including both for consistency, I just thought I'd check because the reasons for one don't apply in userland.

Reply:
Staying consistent seems preferable to me.  Let's just also include a comment explaining the situation for user space.